src/netlify-identity.js

```diff
@@ -141,9 +145,14 @@ observe(store, "siteURL", () => {
   if (store.siteURL === null || store.siteURL === undefined) {
     localStorage.removeItem("netlifySiteURL");
   } else {
+    const currentURL = localStorage.getItem("netlifySiteURL");
```

Comment:
I'm a bit confused by this.
Looking at the code I initially thought we could use the `change` [argument passed to the handler](https://mobx.js.org/refguide/observe.html#observe) instead of relying on `localStorage`, but then realised that it doesn't makes sense for `mobx` to report identical values.

Did a quick test and:
```js
const store = observable({
    siteURL: null,
});
observe(store, "siteURL", (change) => {
    console.log(JSON.stringify(change));
});

store.siteURL = "hello";
store.siteURL = "hello";
```

Only prints a log message on the first assignment to `siteURL`, so I'm not sure how we're getting identical values here. The only thing I can think of is that `localStorage` and `store. siteURL` become out of sync for some reason.

Also, this changes the current behaviour of calling `store.init` when `siteURL` is set to `null` or `undefined`, but I haven't been able to figure if there are any implications to that.

Reply:
@erezrokah as far as I can tell this is only affected by localhost, so I don't _think_ it poses any issues. it's hard to say for sure because I only sort of get how `mobx` works, so I was just testing workflows

maybe we can publish a beta and put it on a few sites to test before doing a full release?